### PR TITLE
xfstests: Solve upload kdump log issue and enable to upload vmcore file

### PR DESCRIPTION
**Pull request by SeroSun**
Sometime during log upload, it will complain folder not exist and cause following upload step not do. This PR fix it in commit1. e.g. http://10.67.133.102/tests/804#step/generate_report/14

Also, in commit2, it enable to upload vmcore and kernel files in test result.

- Related ticket: https://progress.opensuse.org/issues/49868
- Verification run: http://10.67.133.102/tests/805
  Peaceful end and log contain kdump logs.

### CODE REVIEW

**Comment by pevik** [comment]
LGTM. BTW at least some of the global variables in tests/xfstests/run.pm (path related ones, e.g. `$TEST_WRAPPER` are actually constants, so it'd be better to define them as constants, e.g.:
```
use constant {
TEST_WRAPPER => '/usr/share/qa/qa_test_xfstests/wrapper.sh',
STATUS_LOG => '/opt/status.log',
...
};
```

**Comment by SeroSun** [comment]
@pevik indeed, nice catch and thanks! I'll modify it in another PR, since trigger a verify run about kdump take a long time.

**Comment by pevik** [comment]
Sure (didn't meant to postpone this fix).

File: tests/xfstests/generate_report.pm [hunk 1]
```diff
@@ -35,7 +35,7 @@ sub log_end {
 # Compress all sub directories under $dir and upload them.
 sub upload_subdirs {
     my ($dir, $timeout) = @_;
-    my $output = script_output("find $dir -maxdepth 1 -mindepth 1 -type f -or -type d");
+    my $output = script_output("if [ -d $dir ]; then find $dir -maxdepth 1 -mindepth 1 -type f -or -type d; else echo $dir folder not exist; fi");
     for my $subdir (split(/\n/, $output)) {
```
Comment (asmorodskyi) [inline tests/xfstests/generate_report.pm]: so in case folder not exists maybe you don't need to proceed with this for loop ?

Comment (SeroSun) [reply tests/xfstests/generate_report.pm]: Yes, if folder not exist I don't need to upload anything, just print a info to log to say it's not exist.

Comment (SeroSun) [reply tests/xfstests/generate_report.pm]: @asmorodskyi understand where you point out. It's maybe a critical issue, I'll fix before merge.

**Comment by SeroSun** [comment]
New VR would be http://10.67.133.102/tests/812

**Pull request merged**